src/compiler/factory.ts

```diff
@@ -0,0 +1,125 @@
+/// <reference path="core.ts"/>
+/// <reference path="utilities.ts"/>
+
+/* @internal */
+namespace ts {
+    let NodeConstructor: new (kind: SyntaxKind, pos: number, end: number) => Node;
+    let SourceFileConstructor: new (kind: SyntaxKind, pos: number, end: number) => Node;

```

Comment:
This is partly performance related, as the `ts` namespace is horribly deoptimized in v8, but also due to the fact I have a slightly different call signature.